lisa/operating_system.py

```diff
@@ -485,6 +524,36 @@ def get_apt_error(self, stdout: str) -> List[str]:
                 error_lines.append(line)
         return error_lines
 
+    def get_package_information(
+        self, package_name: str, use_cached: bool = True
+    ) -> PackageInformation:
+
+        if use_cached:  # try to fetch cached version
```

Comment:
It can be like below,

```python
package = self._packages.get(package_name, None)

if package and use_cache:
    return package

# get package info
```

Reply:
The use cache is also a sharable logic, it can be in base class, and subclass implement `_get_package_information`